src/bloom.cpp

```diff
@@ -189,21 +190,55 @@ bool CBloomFilter::CheckSpecialTransactionMatchesAndUpdate(const CTransaction &t
         }
         return false;
     }
+    case(TRANSACTION_ASSET_LOCK): {
+        // inputs of Asset Lock transactions are standard. But some outputs are special
+        CAssetLockPayload assetLockTx;
+        if (GetTxPayload(tx, assetLockTx)) {
+            bool fFound = false;
+            const auto& extraOuts = assetLockTx.getCreditOutputs();
+            for (unsigned int i = 0; i < extraOuts.size(); ++i)
+            {
+                fFound = fFound || ProcessTxOut(extraOuts[i], tx.GetHash(), i);
```

Comment:
Should be
```suggestion
                fFound = ProcessTxOut(extraOuts[i], tx.GetHash(), i) || fFound;
```
if we want process all txouts